src/Application.php

```diff
@@ -217,4 +225,56 @@ private function sortConfigRowsByPhase(array $transformationConfigRows): array
 
         return $transformationConfigRows;
     }
+
+    private function checkAndCompletionInputMapping(array $transformationConfig): array
+    {
+        foreach ($transformationConfig['rows'] as $rowKey => $row) {
+            if (!isset($row['configuration']['input'])) {
+                continue;
+            }
+            foreach ($row['configuration']['input'] as $itemKey => $item) {
+                if (!empty($item['datatypes'])) {
+                    try {
+                        $storageTable = $this->storageApiClient->getTable($item['source']);
+                    } catch (Throwable $e) {
+                        continue;
+                    }
+                    $transformationColumnsDatatype = array_map(fn($v) => $v['column'], $item['datatypes']);
+
+                    $missingColumns = array_diff($storageTable['columns'], $transformationColumnsDatatype);
+                    foreach ($missingColumns as $missingColumn) {
+                        if (!isset($storageTable['columnMetadata'][$missingColumn])) {
+                            $newColumn = [
+                                'type' => 'VARCHAR',
+                                'column' => $missingColumn,
+                                'length' => 255,
+                                'convertEmptyValuesToNull' => true,
+                            ];
```

Comment:
Redshift má ještě default `compression: ""` (prázdný string) ale to nevím zda je nutné přidávat.